lib/runtime/PromiseRuntimeModule.js

```diff
@@ -0,0 +1,23 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+*/
+
+"use strict";
+
+const RuntimeGlobals = require("../RuntimeGlobals");
+const RuntimeModule = require("../RuntimeModule");
+
+class PromiseRuntimeModule extends RuntimeModule {
+	constructor() {
+		super("Promise");
+	}
+
+	/**
+	 * @returns {string} runtime code
+	 */
+	generate() {
+		return `${RuntimeGlobals.promise} = typeof Promise !== "undefined" && Promise`;
```

Comment:
In some cases we can optimize the `typeof Promise` away, when we know that the target environment supports promises.

For that add `output.environment.promise` and infer that from version and browserslist. Default to `false` when unknown to avoid breaking change.

Add `RuntimeTemplate.supportPromise()` and check that here.

Reply:
Sure, I'll do it after I fix the CI.